menas/src/main/scala/za/co/absa/enceladus/menas/services/OozieService.scala

```diff
@@ -22,13 +22,41 @@ import org.springframework.stereotype.Component
 
 import za.co.absa.enceladus.menas.models.OozieCoordinatorStatus
 import za.co.absa.enceladus.menas.repositories.OozieRepository
+import za.co.absa.enceladus.model.menas.scheduler.oozie.OozieSchedule
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 
 @Component
 class OozieService @Autowired() (oozieRepository: OozieRepository) {
 
+  import scala.concurrent.ExecutionContext.Implicits.global
+
   def isOozieEnabled: Boolean = oozieRepository.isOozieEnabled()
 
   def getCoordinatorStatus(coordinatorId: String): Future[OozieCoordinatorStatus] = {
     oozieRepository.getCoordinatorStatus(coordinatorId)
   }
+
+  def runNow(oozieSchedule: OozieSchedule, reportDate: Option[String]): Future[String] = {
+    val wfPath = oozieSchedule.activeInstance match {
+      case Some(instance) => instance.workflowPath
+      case None           => throw new IllegalArgumentException("Cannot run a non-active schedule.")
```

Comment:
Can we make this a specific exception for the `ExceptionHandler` to return a proper message to the user?